x-pack/plugins/security_solution/public/common/components/alerts_treemap/lib/layers/index.ts

```diff
@@ -5,38 +5,22 @@
  * 2.0.
  */
 
-import type { Datum } from '@elastic/charts';
+import type { Datum, Key, ArrayNode } from '@elastic/charts';
 
 import { getFillColor } from '../chart_palette';
 import { getLabel } from '../labels';
 
-export interface DataName {
-  dataName: string;
-}
-
-export interface Path {
-  index: number;
-  value: string;
-}
-
-export interface FillColorDatum {
-  path?: Path[];
-}
-
 // common functions used by getLayersOneDimension and getLayersMultiDimensional:
 const valueFormatter = (d: number) => `${d}`;
 const groupByRollup = (d: Datum) => d.key;
 
 /**
  * Extracts the first group name from the data representing the second group
  */
-export const getGroupFromPath = (datum: FillColorDatum): string | undefined => {
+export const getGroupFromPath = (path: ArrayNode['path']): string | undefined => {
   const OFFSET_FROM_END = 2; // The offset from the end of the path array containing the group
-
-  const pathLength = datum.path?.length ?? 0;
-  const groupIndex = pathLength - OFFSET_FROM_END;
-
-  return Array.isArray(datum.path) && groupIndex > 0 ? datum.path[groupIndex].value : undefined;
+  const groupIndex = path.length - OFFSET_FROM_END;
+  return groupIndex > 0 ? path[groupIndex].value : undefined;
```

Comment:
path is always an array, so I've removed all the unnecessary checks.
If you instead have found that this is not true, please open a case in the elastic-charts repo